src/HttpClient.php

```diff
@@ -88,24 +88,69 @@ protected function initializeHandle()
     protected function getBody(RequestInterface $request)
     {
         if (empty($request->getParameters())) {
-            return $request->getBody();
+            return [ "data" => $request->getBody(), "headers" => [] ];
         }
         // special handling for form parameters i.e.
         // returning flatten array with encoded keys if any multipart parameter exists
         // OR returning concatenated encoded parameters string
         $encodedBody = join('&', $request->getEncodedParameters());
         $multipartParameters = $request->getMultipartParameters();
         if (empty($multipartParameters)) {
-            return $encodedBody;
+            return [ 'data' => $encodedBody, 'headers' => [ 'content-type' => 'application/x-www-form-urlencoded' ] ];
         }
         if (empty($encodedBody)) {
-            return $multipartParameters;
+            return $this->createMultiPartFormData($multipartParameters);
         }
         foreach (explode('&', $encodedBody) as $param) {
             $keyValue = explode('=', $param);
             $multipartParameters[urldecode($keyValue[0])] = urldecode($keyValue[1]);
```

Comment:
Since you have updated the structure for each multipart form param, please depict that here. Instead of setting `urldecode($keyValue[1])` directly, set `['data' => urldecode($keyValue[1]), 'headers' => [] ]`